src/webui/api/serialize/serialize_torrent.h

```diff
@@ -94,5 +94,6 @@ inline const QString KEY_TORRENT_AVAILABILITY = u"availability"_s;
 inline const QString KEY_TORRENT_REANNOUNCE = u"reannounce"_s;
 inline const QString KEY_TORRENT_COMMENT = u"comment"_s;
 inline const QString KEY_TORRENT_PRIVATE = u"private"_s;
+inline const QString KEY_TORRENT_HAS_METADATA = u"has_metadata"_s;
```

Comment:
Why "have" and not "has"? While "metadata" is plural, it's the torrent (singular) that has (or does not have) metadata. Thus "has" to me sounds grammatically correct?

Reply:
From https://preply.com/en/blog/has-vs-have/ :
>The main difference between “has” and “have” lies in which subjects they use. “Has” is used with the third-person singular subjects “he,” “she,” and “it,” or a singular noun. “Have” is used with first-person (“I,” “we”), second-person (“you”), and third-person plural (“they”) subjects.

The "metadata" is not a 'third-person' property/subject but a native property of a `torrent` object.

Reply:
I understand that metadata is a native property, but in my view we describe "the torrent", which is the third person singular. It ("the torrent") has properties. 

If the torrent described itself, it would say "I have" properties. But here we are the people describing, thus I'd argue we should say "It has", and in consequence stick with "hasMetadata". It would also be consistent with the torrent data model which uses "hasMetadata()"

What do the others think? @thalieht @stalkerok  @glassez

Reply:
Hmmm after some googling [.NET naming guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members?redirectedfrom=MSDN) uses "has":
>Optionally, you can also prefix Boolean properties with "Is", "Can", or "Has", but only where it adds value.

Well. It seems my opinion is not conclusive. Please disregard it.

Reply:
> Hmmm after some googling [.NET naming guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members?redirectedfrom=MSDN) uses "has":
> 
> > Optionally, you can also prefix Boolean properties with "Is", "Can", or "Has", but only where it adds value.
> 
> Well. It seems my opinion is not conclusive. Please disregard it. Sorry for the noise.

I believe that such naming rules based on ordinary wording, for example, "torrent has metadata".
But Qt naming convention is more abstract from natural language. They require not to use third from for verbs (`is` prefix is special case):
>Verbs have no prefix and don't use the third person (-s)

I suggest not to dwell on it here, since we don't have clear rules about it.